kolibri/core/tasks/job.py

```diff
@@ -156,9 +171,40 @@ def __getstate__(self):
             "args",
             "kwargs",
             "func",
-            "result",
         ]
-        return {key: self.__dict__[key] for key in keys if key in self.__dict__}
+
+        if isinstance(self.exception, Exception):
+            self.exception = type(self.exception).__name__
+
+        if self.result is not None:
+            logger.info(
+                "Please know that results of tasks are currently not saved with jobs."
+            )
+
+        working_dictionary = {
+            key: self.__dict__[key] for key in keys if key in self.__dict__
+        }
+
+        try:
+            string_result = json.dumps(working_dictionary)
+        except TypeError as e:
+            # One or more of the keys is not JSON-serializable.
+            logger.debug("Job '{}' is not be saved: {}".format(self.job_id, str(e)))
```

Comment:
I don't think we should fail silently here - we should raise an exception here to flag that this is not JSON serializable, we should not just carry on.